docs/extensions/resources/extension.md

```diff
@@ -17,11 +17,13 @@ spec:
   resources:
   - kind: Extension
     type: example
-    globallyEnabled: true
+    autoEnable:
+    - shoot
     workerlessSupported: true
 ```
 
-If `spec.resources[].globallyEnabled` is `true`, then the `Extension` resources of the given `type` is created for every shoot cluster. Set to `false`, the `Extension` resource is only created if configured in the `Shoot` manifest. In case of workerless `Shoot`, a globally enabled `Extension` resource is created only if `spec.resources[].workerlessSupported` is also set to `true`. If an extension configured in the spec of a workerless `Shoot` is not supported yet, the admission request will be rejected.
+If `spec.resources[].autoEnable` is set to `shoot`, then the `Extension` resources of the given `type` is created for every shoot cluster. Set to `none` (default), the `Extension` resource is only created if configured in the `Shoot` manifest. In case of workerless `Shoot`, an automatically enabled `Extension` resource is created only if `spec.resources[].workerlessSupported` is also set to `true`. If an extension configured in the spec of a workerless `Shoot` is not supported yet, the admission request will be rejected.
+Other valid values are `seed` (-> automatically enabled for all seeds), as well as `garden` (enabled for shoots, seeds and the garden cluster).
```

Comment:
Hmm, ControllerRegistrations actually do not support the `garden` value right? 
Additionally, this seems a bit confusing wrt https://github.com/gardener/gardener/blob/4bef86bce7fe0229d1490b530b5606fc8eb50537/pkg/apis/operator/v1alpha1/types_garden.go#L704-L705